modules/tensor_mechanics/include/utils/RankTwoScalarTools.h

```diff
@@ -424,69 +424,47 @@ getQuantity(const RankTwoTensorTempl<T> & tensor,
             const Point & curr_point,
             Point & direction)
 {
-  T val = 0.0;
-
   switch (scalar_type)
   {
     case 0:
```

Comment:
ping @dschwen. I was trying to figure out a way to get rid of the ints, but there wasn't a clean way without doing string evaluations (which should be avoided correct?)

Reply:
could use enums...

Reply:
I would have to make a separate enum in the header that matches what is in the .C correct? Not sure if that is more "fool proof" than just making sure the ints match here

Reply:
Yeah, this is really the only place where there is comparisons to the enum (e.g. vonmises, volumetric). It's probably just fine being ints. Doing the return does clean it up nice though, so should be good to go if this was what you were suggesting earlier?